File: moneyExchange/src/main/java/pl/niewadzj/moneyExchange/api/auth/interfaces/AuthService.java
package pl.niewadzj.moneyExchange.api.auth.interfaces;

import pl.niewadzj.moneyExchange.api.auth.records.LoginRequest;
import pl.niewadzj.moneyExchange.api.auth.records.RegistrationRequest;
import pl.niewadzj.moneyExchange.api.auth.records.TokenResponse;

public interface AuthService {

    TokenResponse register(RegistrationRequest registrationRequest);
    TokenResponse login(LoginRequest loginRequest);

}

File: frontend/src/services/authService.js
import {authApi} from "../api/authApi";
import Cookies from 'js-cookie'

export const login = async (loginRequest) => {
    const tokenResponse = await authApi.login(loginRequest);

    if (tokenResponse) {
        Cookies.set("refreshToken", tokenResponse?.data?.refreshToken);
        Cookies.set("authToken", tokenResponse?.data?.authToken);
    }

    return tokenResponse
};

export const register = async (registerRequest) => {
    const tokenResponse = await authApi.register(registerRequest);
    if (tokenResponse) {
        Cookies.set("refreshToken", tokenResponse?.data?.refreshToken);
        Cookies.set("authToken", tokenResponse?.data?.authToken);
    }

    return tokenResponse
};